src/components/AddressBook/SetContactDialog.vue

```diff
@@ -0,0 +1,251 @@
+<template>
+  <dialog-base :title="title" :visible.sync="isVisible" :tooltip="tooltip">
+    <div class="set-address">
+      <s-input
+        ref="address"
+        class="set-address__input"
+        :placeholder="t('nameText')"
+        v-model="name"
+        :disabled="loading"
+        :maxlength="30"
+      />
+      <s-input class="set-address__input" :placeholder="t('addressText')" v-model="address" :disabled="inputDisabled" />
+      <p v-if="isAccountAddress" class="set-address-error">{{ t('walletSend.addressError') }}</p>
+      <template v-if="validAddress && isNotSoraAddress">
+        <p class="wallet-send-address-warning">{{ t('addressBook.notSoraAddress') }}</p>
+        <s-tooltip :content="copyValueAssetId" placement="top">
+          <p class="wallet-send-address-formatted" @click="handleCopyAddress(formattedSoraAddress, $event)">
+            {{ formattedSoraAddress }}
+          </p>
+        </s-tooltip>
+      </template>
+      <s-input
+        class="set-address__input"
+        :placeholder="t('addressBook.identity')"
+        v-model="onChainIdentity"
+        :disabled="true"
+      />
+      <div class="set-address__btn">
+        <s-button @click="setContact" type="primary" class="s-typography-button--large" :disabled="btnDisabled">
+          {{ btnText }}
+        </s-button>
+      </div>
+    </div>
+  </dialog-base>
+</template>
+
+<script lang="ts">
+import debounce from 'lodash/debounce';
+import { Component, Mixins, Prop, Ref, Watch } from 'vue-property-decorator';
+
+import { state, getter, mutation } from '@/store/decorators';
+import type { AccountBook, Book, PolkadotJsAccount } from '@/types/common';
+import { formatSoraAddress } from '@/util';
+
+import { api } from '../../api';
+import DialogBase from '../DialogBase.vue';
+import CopyAddressMixin from '../mixins/CopyAddressMixin';
+import DialogMixin from '../mixins/DialogMixin';
+import LoadingMixin from '../mixins/LoadingMixin';
+import TranslationMixin from '../mixins/TranslationMixin';
+
+@Component({
+  components: {
+    DialogBase,
+  },
+})
+export default class SetContactDialog extends Mixins(DialogMixin, TranslationMixin, LoadingMixin, CopyAddressMixin) {
+  @state.account.book book!: Book;
+  @state.account.polkadotJsAccounts polkadotJsAccounts!: Array<PolkadotJsAccount>;
+  @getter.account.account account!: PolkadotJsAccount;
+  @mutation.account.setAddressToBook setAddressToBook!: (record: AccountBook) => void;
+
+  @Prop({ default: '', type: String }) readonly prefilledAddress!: string;
+  @Prop({ default: false, type: Boolean }) readonly isEditMode!: boolean;
+
+  @Ref('address') private readonly addressInput!: HTMLInputElement;
+
+  @Watch('isVisible')
```

Comment:
do you really need another watcher? you already have `input`/`change` events of s-input component